File: dcp-solutions/2024-11-19/solution.go
```go
package main

import (
	"fmt"
)

func nthPerfectNumber(n int) int {
	count := 0
	num := 1
	for {
		sum := 0
		for num > 0 {
			digit := num % 10
			sum += digit
			num /= 10
		}
		if sum == 10 {
			count++
			if count == n {
				return sum
			}
		}
		num++
	}
}

func main() {
	fmt.Println(nthPerfectNumber(1)) //19
	fmt.Println(nthPerfectNumber(2)) //28
	fmt.Println(nthPerfectNumber(3)) //38
} 
```

File: dcp-solutions/2024-11-19/solution.js
```javascript
function nthPerfectNumber(n) {
    let count = 0;
    let num = 1;
    while (true) {
        if (isPerfect(num)) {
            count++;
            if (count === n) return num;
        }
        num++;
    }
}

function isPerfect(num) {
    let sum = 0;
    for (let digit of num.toString()) {
        sum += parseInt(digit);
    }
    return sum === 10;
}
```